chemprop/v2/cli/train.py

```diff
@@ -497,22 +483,16 @@ def main(args):
     n_tasks = len(args.target_columns)
     bounded = args.loss_function is not None and "bounded" in args.loss_function
 
-    if args.number_of_molecules > 1:
-        warnings.warn(
-            "Multicomponent input is not supported at this time! Using only the 1st input..."
-        )
-
     format_kwargs = dict(
         no_header_row=args.no_header_row,
-        smiles_columns=args.smiles_columns,
-        target_columns=args.target_columns,
+        smiles_cols=args.smiles_columns,
+        rxn_cols=args.reaction_columns,
         bounded=bounded,
     )
     featurization_kwargs = dict(
         features_generators=args.features_generators,
         keep_h=args.keep_h,
         add_h=args.add_h,
-        reaction=0 in args.rxn_idxs,  # TODO: check if this is correct
```

Comment:
I think we still need something like `reaction=not args.reaction_columns` to check that this is not a reaction as it affects the kind of featurizer to use down the line.

Reply:
We don't. We can infer the featurizer needed based on the type of the data fed into `make_dataset()`:
```python
def make_dataset(
    data: Sequence[MoleculeDatapoint] | Sequence[ReactionDatapoint],
    bond_messages: bool,
    reaction_mode: str,
) -> MoleculeDataset | ReactionDataset:
    if isinstance(data[0], MoleculeDatapoint):
        extra_atom_fdim = data[0].V_f.shape[1] if data[0].V_f is not None else 0
        extra_bond_fdim = data[0].E_f.shape[1] if data[0].E_f is not None else 0
        featurizer = MoleculeMolGraphFeaturizer(
            bond_messages=bond_messages,
            extra_atom_fdim=extra_atom_fdim,
            extra_bond_fdim=extra_bond_fdim,
        )
        return MoleculeDataset(data, featurizer)


    featurizer = CondensedGraphOfReactionFeaturizer(
        bond_messages=bond_messages, mode_=reaction_mode
    )


    return ReactionDataset(data, featurizer)
```